libass/ass_fontconfig.c

```diff
@@ -88,14 +88,25 @@ static void destroy(void *priv)
     free(fc);
 }
 
-static void scan_fonts(FcConfig *config, ASS_FontProvider *provider)
+static void scan_fonts(FcConfig *config, ASS_FontProvider *provider, ASS_Library *lib)
 {
     int i;
     FcFontSet *fonts;
     ASS_FontProviderMetaData meta = {0};
 
-    // get list of fonts
-    fonts = FcConfigGetFonts(config, FcSetSystem);
+    // get list of fonts;
+    // sorting by default pattern prefers regular variants
+    FcPattern *pat = FcPatternCreate();
+    FcDefaultSubstitute(pat);
+    FcResult res;
+    // trim=FcFalse returns all system fonts
+    fonts = FcFontSort(config, pat, FcFalse, NULL, &res);
```

Comment:
Can this return `NULL` without also setting `res != FcResultMatch`?

Reply:
I’d expect no, since why else have result return value and looking at current code seems to support this. However one peculiar comment suggests at least the underlying `FcFontSetSort` may have done so in the past:
```C
 /* There are some implementation that relying on the result of
 * "result" to check if the return value of FcFontSetSort
 * is valid or not.
 * So we should initialize it to the conservative way since
 * this function doesn't return NULL anymore.
 */
```

Reply:
This comment was introduced in [fontconfig@a18ca17b](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/commit/a18ca17b6211f62fbd1d893811b94b8c83db4cc0) and released with 2.9.0 *(twelve years ago)*.
We require fontconfig `>= 2.10.92` and therefore can rely on `res`.